src/sinks/datadog/mod.rs

```diff
@@ -45,49 +41,89 @@ pub struct DatadogCommonConfig {
     #[configurable(metadata(docs::examples = "http://127.0.0.1:8080"))]
     #[configurable(metadata(docs::examples = "http://example.com:12345"))]
     #[serde(default)]
-    pub endpoint: Option<String>,
+    endpoint: Option<String>,
 
     /// The Datadog [site][dd_site] to send observability data to.
     ///
+    /// If not set here, the value is taken from the [global options][global_options].
+    ///
     /// [dd_site]: https://docs.datadoghq.com/getting_started/site
+    /// [global_options]: /docs/reference/configuration/global-options/#datadog
     #[configurable(metadata(docs::examples = "us3.datadoghq.com"))]
     #[configurable(metadata(docs::examples = "datadoghq.eu"))]
-    #[serde(default = "default_site")]
-    pub site: String,
+    site: Option<String>,
```

Comment:
Thanks to the new design, we can make a few adjustments to the docs for `site` and `default_api_key`.

```
    /// The Datadog [site][dd_site] to send observability data to.
    ///
    /// This value can also be set by specifying the `DD_SITE` environment variable.
    /// The value specified here takes precedence over the environment variable.
    ///
    /// [dd_site]: https://docs.datadoghq.com/getting_started/site
    #[configurable(metadata(docs::examples = "us3.datadoghq.com"))]
    #[configurable(metadata(docs::examples = "datadoghq.eu"))]
    #[serde(default = "default_site")]
    site: Option<String>,
```

```
    /// The default Datadog [API key][api_key] to use in authentication of HTTP requests.
    ///
    /// If an event has a Datadog [API key][api_key] set explicitly in its metadata, it takes
    /// precedence over this setting.
    ///
    /// This value can also be set by specifying the `DD_API_KEY` environment variable.
    /// The value specified here takes precedence over the environment variable.
    ///
    /// [api_key]: https://docs.datadoghq.com/api/?lang=bash#authentication
    #[configurable(metadata(docs::examples = "${DATADOG_API_KEY_ENV_VAR}"))]
    #[configurable(metadata(docs::examples = "ef8d5de700e7989468166c40fc8a0ccd"))]
    default_api_key: Option<SensitiveString>,
```